packages/@aws-cdk/toolkit-lib/test/actions/deploy-hotswap.test.ts

```diff
@@ -33,42 +32,82 @@ beforeEach(() => {
 });
 
 describe('deploy with hotswap', () => {
-  test('does print hotswap warnings for FALL_BACK mode', async () => {
+  test('does print hotswap warnings for hotswap with fallback', async () => {
     // WHEN
     const cx = await builderFixture(toolkit, 'two-empty-stacks');
     await toolkit.deploy(cx, {
-      hotswap: HotswapMode.FALL_BACK,
+      deploymentMethod: {
+        method: 'hotswap',
+        fallback: { method: 'change-set' },
+      },
```

Comment:
So previously the fallback method would be derived from the default of `deploymentMethod` - right? And now it seems like if a user wants to hotswap with fallback, they **must** also provide the fallback method as well. This is a slightly more verbose API and I wonder if we can instead have:

```suggestion
      deploymentMethod: {
        method: 'hotswap',
        fallback: true, // default to change-set
        // if a user wants
        fallbackMethod: 'direct'
      },
```

Also not super elegant. Your call.

Reply:
I thought about this. In the CLI if you run `cdk deploy --hotswap` it does hotswap-only aka no fallback. To get the fallback you have to pass `cdk deploy --hotswap-fallback`. You cannot pass both flags at the same time. `cdk watch` defaults to hotswap-only for all deployments unless `cdk watch --no-hotswap` is passed.

So really the CLI doesn't have a default but a slight bias towards hotswap-only, it being the more prominent name and the watch default.

This was replicated in the toolkit-lib with the `HotswapMode` enum. You had to make an explicit choice, but you got the style of fallback method for free from the default for `deploymentMethod`.

---

The problem with your suggestion is that `deploymentMethod` has additional parameters that can be set, so it's not enough to just have a single `fallbackMethod` field.

I considered defaulting fallback to `{ method: 'change-set' }` and introduce something like `{ method: 'none' }` but that is against the existing bias towards no fallback and also pollutes the `DeploymentMethod` type with a none variant.

Another option is the introduction of enum-like helpers: `DeploymentMethod.changeSet()` and `DeploymentMethod.direct()` etc. which makes this slightly nicer to write. We do this elsewhere but ultimately I thought it's not needed here. This can be added afterwards as well.